File: sdk-source/typescript/examples/quickstart.ts
/**
 * Runnable quickstart for the Aegis TypeScript SDK.
 *
 *   AEGIS_BASE_URL=https://your-aegis-host AEGIS_APP_KEY=pk_live_... npx tsx examples/quickstart.ts
 */
import { Aegis, AegisError } from "../src/index";

const aegis = new Aegis({
  baseUrl: process.env.AEGIS_BASE_URL ?? "http://localhost:8080",
  appKey: process.env.AEGIS_APP_KEY ?? "",
  version: "1.0.0",
});

async function main() {
  const init = await aegis.init();
  console.log("Initialized:", init.status);

  if (init.version?.update_required) {
    console.log("Mandatory update:", init.version.latest, init.version.download_url);
    return;
  }

  const username = process.env.AEGIS_USERNAME ?? "demo";
  const password = process.env.AEGIS_PASSWORD ?? "demo-password";

  const { user, license } = await aegis.login({ username, password });
  console.log("Signed in as", user.username, "license:", license?.status ?? "none");

  const vars = await aegis.getVariables("user");
  console.log("User variables:", vars.variables);
  await aegis.setVariable("last_seen", new Date().toISOString());

  const stop = aegis.startHeartbeat({
    intervalMs: 60_000,
    onRevoked: (reason) => console.warn("Session revoked:", reason),
  });

  await new Promise((resolve) => setTimeout(resolve, 2_000));
  stop();
  await aegis.logout();
  console.log("Signed out.");
}

main().catch((error) => {
  if (error instanceof AegisError) {
    console.error(`Aegis error [${error.code}] ${error.message}`);
    process.exit(1);
  }
  throw error;
});

File: sdk-source/python/examples/quickstart.py
"""Runnable quickstart for the Aegis Python SDK.

    AEGIS_BASE_URL=https://your-aegis-host AEGIS_APP_KEY=pk_live_... python examples/quickstart.py
"""
import os
import sys
import time

sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))

from aegis import Aegis, AegisError  # noqa: E402


def main() -> int:
    client = Aegis(
        base_url=os.environ.get("AEGIS_BASE_URL", "http://localhost:8080"),
        app_key=os.environ.get("AEGIS_APP_KEY", ""),
        version="1.0.0",
    )

    info = client.init()
    print("Initialized:", info.get("status"))

    version = info.get("version") or {}
    if version.get("update_required"):
        print("Mandatory update:", version.get("latest"), version.get("download_url"))
        return 0

    result = client.login(
        os.environ.get("AEGIS_USERNAME", "demo"),
        os.environ.get("AEGIS_PASSWORD", "demo-password"),
    )
    license_info = result.get("license") or {}
    print("Signed in as", result["user"]["username"], "license:", license_info.get("status", "none"))

    print("User variables:", client.get_variables("user").get("variables"))
    client.set_variable("last_seen", time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime()))

    with client.start_heartbeat(interval=60, on_revoked=lambda reason: print("Session revoked:", reason)):
        time.sleep(2)

    client.logout()
    print("Signed out.")
    return 0


if __name__ == "__main__":
    try:
        raise SystemExit(main())
    except AegisError as error:
        print(f"Aegis error [{error.code}] {error.message}", file=sys.stderr)
        raise SystemExit(1)